androidApp/src/main/java/com/example/tasky/android/alarm/data/SyncAgendaManager.kt

```diff
@@ -34,10 +34,11 @@ class SyncAgendaManager(
             ).build()
 
     override fun startPeriodicSyncAgenda() {
+        workManager.cancelAllWorkByTag(SyncAgendaWorker.TAG)
         workManager.enqueue(workRequest)
     }
 
     override fun stopPeriodicSyncAgenda() {
-        workManager.cancelWorkById(workRequest.id)
+        workManager.cancelAllWorkByTag(SyncAgendaWorker.TAG)
```

Comment:
Minor - you could also call `stopPeriodicSyncAgenda()` inside the `startPeriodicSyncAgenda()` function instead of duplicating the logic to cancel the workers.